### PR TITLE
sharness: Init at 1.1.0-dev

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Motivation for this change

A simple portable shell library to write and run tests adhering to the Test Anything Protocol.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/sharness/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+, lib
+, fetchFromGitHub
+, fetchurl
+, perl
+, perlPackages
+, sharnessExtensions ? {} }:
+let
+  inherit (lib) mapAttrsToList optionalString;
+
+in stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:11]:
```suggestion

stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/sharness/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  # Used for testing
+  nativeBuildInputs = [ perl perlPackages.IOTty ];
+
+  outputs = [ "out" "doc" ];
+
+  makeFlags = [ "prefix=$(out)" ];
+
+  extensions = mapAttrsToList (k: v: "${k}.sh ${v}") sharnessExtensions;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:28]:
```suggestion
  extensions = lib.mapAttrsToList (k: v: "${k}.sh ${v}") sharnessExtensions;
```

File: pkgs/development/libraries/sharness/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  postInstall = optionalString (sharnessExtensions != {}) ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:30]:
```suggestion
  postInstall = lib.optionalString (sharnessExtensions != {}) ''
```

File: pkgs/development/libraries/sharness/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    linkExtensions
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Portable shell library to write, run and analyze automated tests adhering to Test Anything Protocol (TAP)";
+    homepage = "https://github.com/chriscool/sharness";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:48]: gpl2Only or gpl2Plus?

File: pkgs/development/libraries/sharness/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+, perl
+, perlPackages
+, sharnessExtensions ? {} }:
+let
+  inherit (lib) mapAttrsToList optionalString;
+
+in stdenv.mkDerivation rec {
+  pname = "sharness";
+  version = "1.1.0-dev";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:14]:
```suggestion
  version = "1.1.0-dev";

  src = fetchFromGitHub {
```

**Comment by spacefrogg** [comment]
Updated.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112309` at 44b06c61 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/689f316f543f91a84219720f1fc63a58">sharness</a></li></li>
  </ul>
</details>

**Pull request merged**